src/ucp/tag/eager_snd.c

```diff
@@ -214,11 +232,28 @@ static ucs_status_t ucp_tag_eager_sync_bcopy_single(uct_pending_req_t *self)
 
 static ucs_status_t ucp_tag_eager_sync_bcopy_multi(uct_pending_req_t *self)
 {
-    ucs_status_t status = ucp_do_am_bcopy_multi(self,
-                                                UCP_AM_ID_EAGER_SYNC_FIRST,
-                                                UCP_AM_ID_EAGER_MIDDLE,
-                                                ucp_tag_pack_eager_sync_first_dt,
-                                                ucp_tag_pack_eager_middle_dt, 1);
+    ucp_request_t *req = ucs_container_of(self, ucp_request_t, send.uct);
+    size_t length;
+    ucs_status_t status;
+
+    if (ucs_unlikely(ucp_wireup_ep_test(req->send.ep->uct_eps[req->send.lane]))) {
+        return UCS_ERR_NO_RESOURCE;
+    }
+
+    req->send.lane = ucp_ep_get_am_lane(req->send.ep);
+    length = ucs_min(ucp_ep_get_max_bcopy(req->send.ep, req->send.lane) -
+                                          sizeof(ucp_eager_sync_hdr_t),
+                     req->send.length);
+    if (ucs_unlikely(length <= req->send.length)) {
+        /* fallback to single packet due to EP reconfiguration after wireup */
+        req->send.uct.func = ucp_tag_eager_sync_bcopy_single;
+        return ucp_tag_eager_sync_bcopy_single(self);
+    }
```

Comment:
what error does it provoke? I'd expect that if single eager messages arrives as a "first eager" fragment, it is correctly handled here: https://github.com/openucx/ucx/blob/master/src/ucp/tag/eager_rcv.c#L112

Reply:
first and single packet is wrongly added to frag list. The fix depends on how it's assumed by design, if that's OK to handle it on recv then need to check return code there, otherwise need to avoid sending to "wrong" AM handler like it's implemented in this PR now